src/Nethermind/Nethermind.Core2/Containers/BeaconBlockBody.cs

```diff
@@ -0,0 +1,130 @@
+//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using Nethermind.Core.Extensions;
+using Nethermind.Core2.Crypto;
+
+namespace Nethermind.Core2.Containers
+{
+    public class BeaconBlockBody
+    {
+        public bool Equals(BeaconBlockBody other)
+        {
+            bool basicEquality = Equals(RandaoReversal, other.RandaoReversal) &&
+                                 Equals(Eth1Data, other.Eth1Data) &&
+                                 Bytes.AreEqual(Graffiti, other.Graffiti) &&
+                                 ProposerSlashings.Length == other.ProposerSlashings.Length &&
+                                 AttesterSlashings.Length == other.AttesterSlashings.Length &&
+                                 Attestations.Length == other.Attestations.Length &&
+                                 Deposits.Length == other.Deposits.Length &&
+                                 VoluntaryExits.Length == other.VoluntaryExits.Length;
+
+            if (!basicEquality)
+            {
+                return false;
+            }
+            
+            for (int i = 0; i < AttesterSlashings.Length; i++)
+            {
+                if (!Equals(AttesterSlashings[i], other.AttesterSlashings[i]))
+                {
+                    return false;
+                }
+            }
```

Comment:
This can be done with generic method for arrays.

Reply:
probably, worth trying with the approach where SszLength is an instance property
then we can use generics for arrays - I have been already thinking about it.

===

not sure how critical will be performance here, give it some time - need to write other things first